crates/biome_js_analyze/src/lint/nursery/use_google_font_display.rs

```diff
@@ -0,0 +1,137 @@
+use biome_analyze::{
+    context::RuleContext, declare_lint_rule, Ast, Rule, RuleDiagnostic, RuleSource, RuleSourceKind,
+};
+use biome_console::markup;
+use biome_js_syntax::jsx_ext::AnyJsxElement;
+use biome_rowan::TextRange;
+
+declare_lint_rule! {
+    /// Enforces the use of a recommended `font-display` strategy with Google Fonts.
+    ///
+    /// The `font-display` property controls how a font is displayed while it is loading. When using Google Fonts,
+    /// it's important to specify an appropriate value for this property to ensure good user experience and prevent layout shifts.
+    ///
+    /// This rule flags the absence of the `font-display` parameter, or the usage of less optimal values such as `auto`, `block`, or `fallback`.
+    /// Using `&display=optional` is generally recommended as it minimizes the risk of invisible text or layout shifts.
+    /// In cases where swapping to the custom font after it has loaded is important, consider using `&display=swap`.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <link href="https://fonts.googleapis.com/css2?family=Krona+One" />
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <link href="https://fonts.googleapis.com/css2?family=Krona+One&display=auto" />
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <link href="https://fonts.googleapis.com/css2?family=Krona+One&display=block" />
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <link href="https://fonts.googleapis.com/css2?family=Krona+One&display=fallback" />
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```jsx
+    /// <link href="https://fonts.googleapis.com/css2?family=Krona+One&display=optional" rel="stylesheet" />
+    /// ```
+    ///
+    /// ```jsx
+    /// <link href="https://fonts.googleapis.com/css2?display=unknown" rel="stylesheet" />
+    /// ```
+    ///
+    /// ```jsx
+    /// <link rel="stylesheet" />
+    /// ```
+    pub UseGoogleFontDisplay {
+        version: "next",
+        name: "useGoogleFontDisplay",
+        language: "jsx",
+        sources: &[RuleSource::EslintNext("google-font-display")],
+        source_kind: RuleSourceKind::SameLogic,
+        recommended: false,
+    }
+}
+
+impl Rule for UseGoogleFontDisplay {
+    type Query = Ast<AnyJsxElement>;
+    type State = (bool, TextRange);
```

Comment:
Can you save the state into a variable a document it? You'll have to document especially what the `bool` is. It can also be a type:

```rs
type RuleState = (bool, TextRange);
```